qiskit/optimization/problems/quadratic_program.py

```diff
@@ -0,0 +1,1101 @@
+        if self.objective.sense == QuadraticObjective.Sense.MINIMIZE:
+            mdl.minimize(objective)
+        else:
+            mdl.maximize(objective)
+
+        # add linear constraints
+        for i, constraint in enumerate(self.linear_constraints):
+            name = constraint.name
+            rhs = constraint.rhs
+            if rhs == 0 and constraint.linear.coefficients.nnz == 0:
+                continue
+            linear_expr = 0
+            for j, v in constraint.linear.to_dict().items():
+                linear_expr += v * var[j]
+            sense = constraint.sense
+            if sense == Constraint.Sense.EQ:
+                mdl.add_constraint(linear_expr == rhs, ctname=name)
+            elif sense == Constraint.Sense.GE:
+                mdl.add_constraint(linear_expr >= rhs, ctname=name)
+            elif sense == Constraint.Sense.LE:
+                mdl.add_constraint(linear_expr <= rhs, ctname=name)
+            else:
+                # should never happen
+                raise QiskitOptimizationError("Unsupported constraint sense: {}".format(sense))
+
+        # add quadratic constraints
+        for i, constraint in enumerate(self.quadratic_constraints):
+            name = constraint.name
+            rhs = constraint.rhs
+            if rhs == 0 \
+                    and constraint.linear.coefficients.nnz == 0 \
+                    and constraint.quadratic.coefficients.nnz == 0:
+                continue
+            quadratic_expr = 0
+            for j, v in constraint.linear.to_dict().items():
+                quadratic_expr += v * var[j]
+            for (j, k), v in constraint.quadratic.to_dict().items():
+                quadratic_expr += v * var[j] * var[k]
+            sense = constraint.sense
+            if sense == Constraint.Sense.EQ:
+                mdl.add_constraint(quadratic_expr == rhs, ctname=name)
+            elif sense == Constraint.Sense.GE:
+                mdl.add_constraint(quadratic_expr >= rhs, ctname=name)
+            elif sense == Constraint.Sense.LE:
+                mdl.add_constraint(quadratic_expr <= rhs, ctname=name)
+            else:
+                # should never happen
+                raise QiskitOptimizationError("Unsupported constraint sense: {}".format(sense))
+
+        return mdl
+
+    def pprint_as_string(self) -> str:
+        """Pretty prints the quadratic program as a string.
+
+        Returns:
+            A string representing the quadratic program.
+        """
+        return self.to_docplex().pprint_as_string()
+
+    def prettyprint(self, out: Optional[str] = None) -> None:
+        """Pretty prints the quadratic program to a given output stream (None = default).
+
+        Args:
+            out: The output stream or file name to print to.
+              if you specify a file name, the output file name is has '.mod' as suffix.
+        """
+        self.to_docplex().prettyprint(out)
+
+    def print_as_lp_string(self) -> str:
```

Comment:
With the above methods they seem to be the same as the docplex model names. This one, and below, I see different names. Is there a reason in these to change the names or if you were used to working with docplex model would it be better to keep the same names like the above functions.

Also this method just returns a string and does not print it despite its name

Reply:
Though I agree that something saying `print` should actually print, the `_as_string` suffix was made to match docplex, where this means a string will be returned. I updated the documentation saying "returns a.. " instead of "prints... "